konnectivity-client/pkg/client/client_test.go

```diff
@@ -59,6 +59,49 @@ func TestDial(t *testing.T) {
 	}
 }
 
+func TestDialRace(t *testing.T) {
+	ctx := context.Background()
+	s, ps := pipe()
+	ts := testServer(ps, 100)
+
+	defer ps.Close()
+	defer s.Close()
+
+	tunnel := &grpcTunnel{
+		// artificially delay after calling Send, ensure handoff of result from serve to DialContext still works
+		stream:      fakeSlowSend{s},
+		pendingDial: make(map[int64]pendingDial),
+		conns:       make(map[int64]*conn),
+	}
+
+	go tunnel.serve(&fakeConn{})
+	go ts.serve()
+
+	_, err := tunnel.DialContext(ctx, "tcp", "127.0.0.1:80")
+	if err != nil {
+		t.Fatalf("expect nil; got %v", err)
+	}
+
+	if ts.packets[0].Type != client.PacketType_DIAL_REQ {
+		t.Fatalf("expect packet.type %v; got %v", client.PacketType_CLOSE_REQ, ts.packets[0].Type)
```

Comment:
```suggestion
	if ts.packets[0].Type != client.PacketType_DIAL_REQ {
		t.Fatalf("expect packet.type %v; got %v", client.PacketType_DIAL_REQ, ts.packets[0].Type)
```

Reply:
ha, this was a copy/paste of TestDial... should be fixed in both